aten/src/ATen/cpu/vec/vec512/vec512_base.h

```diff
@@ -0,0 +1,910 @@
+#pragma once
+
+// DO NOT DEFINE STATIC DATA IN THIS HEADER!
+// See Note [Do not compile initializers with AVX]
+//
+// Note [Do not compile initializers with AVX]
+// ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+// If you define a static initializer in this file, the initialization will use
+// AVX instructions because these object files are compiled with AVX enabled.
+// We need to avoid non-trivial global data in these architecture specific files
+// because there's no way to guard the global initializers with CPU capability
+// detection.
+//
+// See https://github.com/pytorch/pytorch/issues/37577 for an instance
+// of this bug in the past.
+
+#include <cstring>
+#include <functional>
+#include <cmath>
+#include <type_traits>
+#include <bitset>
+
+#include <ATen/cpu/vec/vec512/intrinsics.h>
+#include <ATen/native/Math.h>
+#include <ATen/NumericUtils.h>
+#include <c10/util/C++17.h>
+#include <c10/util/BFloat16.h>
+#include <c10/util/BFloat16-math.h>
+#include <c10/util/copysign.h>
+#include <c10/util/math_compat.h>
+#include <ATen/native/cpu/zmath.h>
+#include <c10/util/TypeCast.h>
+#include <c10/macros/Macros.h>
+
+#if defined(__GNUC__)

```

Comment:
Is __GNUC__ defined for both clang and gcc? Or clang variables do not need to be 64-byte aligned?

Reply:
Yes, it's defined for both GCC & clang.

Reply:
Thanks, I realized there was a `clang` specific check in `intrinsics.h`, so I removed it as well.